static/js/components/ViewProfileDisplay.js

```diff
@@ -0,0 +1,210 @@
+
+import type { Country, CurrentUser } from "../flow/authTypes"
+import { setDrawerState } from "../reducers/drawer"
+
+type StateProps = {|
+  currentUser: ?CurrentUser,
+  countries: Array<Country>
+|}
+
+type DispatchProps = {
+  updateDrawer(state): void
+}
+
+type Props = {|
+  ...StateProps,
+  ...DispatchProps
+|}
+
+export function ViewProfileDisplay(props: Props) {
+  const { currentUser, countries, updateDrawer } = props
+
+  return countries && currentUser ? (
+    <div className="container auth-page registration-page">
+      <MetaTags>
+        <title>{formatTitle(VIEW_PROFILE_PAGE_TITLE)}</title>
+      </MetaTags>
+      <div className="auth-header row">
+        <h1 className="col-6">Profile</h1>
+        <div className="col-6 profile-button-col">
+          <button
+            type="submit"
+            onClick={() => {
+              updateDrawer("profileEdit")
+            }}
+            className="btn btn-danger profile-btn"
+          >
+            Edit
+          </button>
+        </div>
+      </div>
+      <div className="auth-card card-shadow row">
+        <div className="container profile-container ">
+          <div className="row">
+            {currentUser.is_authenticated ? (
+              <div className="col-12 auth-form">
+                <div className="row profile-row">
+                  <div className="col profile-label">First Name</div>
+                  <div className="col">
+                    {currentUser.legal_address.first_name}
+                  </div>
+                </div>
+                <div className="row profile-row">
+                  <div className="col profile-label">Last Name</div>
+                  <div className="col">
+                    {currentUser.legal_address.last_name}
+                  </div>
+                </div>
+                <div className="row profile-row">
+                  <div className="col profile-label">Full Name</div>
+                  <div className="col">{currentUser.profile.name}</div>
+                </div>
+                {currentUser.legal_address.street_address.map((line, idx) => (
+                  <div className="row profile-row" key={idx}>
+                    <div className="col profile-label">Address {idx + 1}</div>
+                    <div className="col">{line}</div>
+                  </div>
+                ))}
+                {countries && currentUser.legal_address.country ? (
+                  <div className="row profile-row">
+                    <div className="col profile-label">Country</div>
+                    <div className="col">
+                      {
+                        find(
+                          propEq("code", currentUser.legal_address.country),
+                          countries
+                        ).name
+                      }
+                    </div>
+                  </div>
+                ) : null}
+                {countries && currentUser.legal_address.state_or_territory ? (
```

Comment:
The design shows a city field between the country and state fields. Is that information available here?

Reply:
yes, removed by accident